docs/source/data_extraction.rst

```diff
@@ -0,0 +1,83 @@
+.. data_extraction:
+
+Data Extraction
+===============
+
+Ara is a component of the Citrine Platform data service that extracts data from GEMD's complex and expressive graphical representations into a tabular structure (like a CSV file) that is easier to consume in analytical contexts.  The rows in the resulting Ara Table are in 1-to-1 correspondence with the set of material histories.  Columns correspond to data about the material histories, such as the temperature measured in a kiln used at a specific manufacturing step.
+
+Defining row and columns
+------------------------
+
+A Row object describes a mapping from a dataset to rows of a table. The only such mapping presently supported is :class:`~citrine.ara.rows.MaterialRunByTemplate`, which produces one row per Material Run associated with any of a list of material templates.
+
+.. code-block:: python
+
+   from citrine.ara.rows import MaterialRunByTemplate
+   row_def = MaterialRunByTemplate(
+         templates=[LinkByUID(scope="templates", id="finished cookie")])
+
+A :class:`~citrine.ara.variables.Variable` object specifies how to select a piece of data from each Material History. Thus, it serves as a mapping from the set of Material Histories to a column in the Ara table.
+
+.. code-block:: python
+
+   from citrine.ara.variables import AttributeByTemplate
+   final_density = AttributeByTemplateAfterProcessTemplate(
+         name = "final density",
+         attribute_template = LinkByUID(scope="templates", id="cookie density"),
+         process_template = LinkByUID(scope="templates", id="apply glaze")
+
+A :class:`~citrine.ara.columns.Column` object describes how to transform a Variable into a primitive value (e.g. a real number, an integer, or a string) that can be entered into a table.  This is necessary because GEMD Attributes are more general than primitive values; they often conveying uncertainty estimates, for example.
```

Comment:
```suggestion
A :class:`~citrine.ara.columns.Column` object describes how to transform a Variable into a primitive value (e.g. a real number, an integer, or a string) that can be entered into a cell in a table.
This is necessary because GEMD Attributes are more general than primitive values; they often conveying uncertainty estimates, for example.
```